source/mainnet/smart-contracts/tutorials/piggy-bank/deploying.rst

```diff
@@ -0,0 +1,160 @@
+
+.. code-block:: console
+
+   $cargo concordium build
+
+.. Note::
+   This command creates a module ``piggy_bank_part2.wasm.v1`` in the folder ``./target/concordium/wasm32-unknown-unknown/release/``.
+   Please use the wasm module file suffixed with ``v1`` when deploying to the chain. The ``piggy_bank_part2.wasm`` file (without the ``v1`` suffix) is not the ``v0`` module, but rather the raw wasm module produced by cargo.
+   The ``v1`` suffixed module has been stripped for debugging information, which makes it significantly smaller. Modules already include the smart contract version (and the size of the module).
+
+
+The ``piggy_bank_part2.wasm.v1`` file can be deployed to the Concordium chain. Let’s transfer the ``piggy_bank_part2.wasm.v1`` file to the place where you are running your concordium-client tool. Move to the folder as well.
+
+Deploying a module
+==================
+You are now set up to deploy the piggy bank module to the testnet chain. Let's give the module the name ``piggy_bank_part2_module`` so it can be referenced in this tutorial.
+
+.. Note::
+   Please remember to adjust the following commands when you are running your ``concordium-client`` tool locally and want to connect to your node on the server. You can look up the different options on how to connect the ``concordium-client`` tool to your testnet node :ref:`here<interacting-with-your-testnet-node>`.
+
+.. code-block:: console
+
+   $./concordium-client module deploy piggy_bank_part2.wasm.v1 --sender <account-name> --name piggy_bank_part2_module --grpc-port 10001
+
+
+.. Note::
+   This tutorial uses ``Concordium`` for the <account-name>.
+
+.. Note::
+   You will be asked to input a password. Use the ``export password`` that you used for creating the key backup file.
+
+If everything works correctly the output has a green line with your module reference. You can find additional information about deploying a module in this :ref:`guide<deploy-module>`.
+
+.. code-block:: console
+
+   Module successfully deployed with reference: <module-reference>.
+   Module reference <module-reference> was successfully named 'piggy_bank_part2_module'.
+
+
+Initializing a module
+=====================
+
+You can initialize your smart contract now to create a smart contract instance. Let's give the instance the name ``piggy_bank_part2_instance`` so it can be referenced in this tutorial.
+
+.. code-block:: console
+
+   $./concordium-client contract init piggy_bank_part2_module --sender <account-name> --contract PiggyBank --name piggy_bank_part2_instance --energy 1000 --grpc-port 10001
+
+.. Note::
+   1000 is enough energy to initialize the piggy bank smart contract but you might need more energy for larger smart contracts.
```

Comment:
I would also add something like: `The --energy parameter sets the maximum allowed energy to be used. You only pay for the actual energy used.`